cpp/src/search/contains_nested.cu

```diff
@@ -59,4 +65,173 @@ bool contains_nested_element(column_view const& haystack,
+  table_comparator_adapter(Comparator&& comp_) : comp(std::move(comp_)) {}
+
+  /**
+   * @brief Call the comparator with strong index types from the row indices of size_type type.
+   *
+   * Given two row indices `i` and `j`, this functor converts the non-negative value into
+   * `lhs_index_type` and the negative value into `rhs_index_type`. Before such conversion
+   * happening, it also converts the negative index into its valid range using `index_adapter`
+   * functor.
+   *
+   * Note that there is not any validity check to be performed in this adapter function. Caller is
+   * responsible to make sure the input parameters are valid: one index must be non-negative and
+   * the other must be negative.
+   *
+   * @param i Index of a row in one table to compare.
+   * @param j Index of a row in the other table to compare.
+   * @return The row comparison result.
+   */
+  __device__ inline auto operator()(size_type const i, size_type const j) const noexcept
+  {
+    using cudf::experimental::row::lhs_index_type;
+    using cudf::experimental::row::rhs_index_type;
+
+    auto const lhs_idx = static_cast<lhs_index_type>(i >= 0 ? i : j);
+    auto const rhs_idx = static_cast<rhs_index_type>(index_adapter{}(i < 0 ? i : j));
+
+    return comp(lhs_idx, rhs_idx);
+  }
+
+ private:
+  Comparator const comp;
+};
+
+/**
+ * @brief The adapter struct for calling a row hasher from negative row indices.
+ *
+ * @tparam RowHasher A class of row hasher that supports nested types data.
+ */
+template <typename RowHasher>
+struct row_hasher_adapter {
+  row_hasher_adapter(RowHasher&& hasher_) : hasher(std::move(hasher_)) {}
+
+  /**
+   * Given a negative row index `i`, this functor converts the index into its valid range
+   * using `index_adapter` functor before calling to the underlying row hasher.
+   *
+   * The will not be any validity check for the input value `i`, assuming that it is negative.
+   *
+   * @param i A row index that is given as a negative value.
+   * @return The resulting hash value.
+   */
+  __device__ inline auto operator()(size_type const i) const noexcept
+  {
+    return hasher(index_adapter{}(i));
+  }
+
+ private:
+  RowHasher const hasher;
+};
+
+}  // namespace
+
+std::unique_ptr<column> multi_contains_nested_elements(column_view const& haystack,
+                                                       column_view const& needles,
+                                                       rmm::cuda_stream_view stream,
+                                                       rmm::mr::device_memory_resource* mr)
+{
+  auto result = make_numeric_column(data_type{type_to_id<bool>()},
+                                    needles.size(),
+                                    copy_bitmask(needles),
+                                    needles.null_count(),
+                                    stream,
+                                    mr);
+  if (needles.is_empty()) { return result; }
+  (void)haystack;
```

Comment:
Why is this needed? The parameter is used below in `if (haystack.is_empty())` so I don't think the cast-to-void is needed to prevent unused parameter warnings.